apps/passport-server/migrations/37_add_encryption_key_column.sql

```diff
@@ -0,0 +1,5 @@
+-- Add column for storing a user's encryption key for their data,
+-- if they elect to skip setting up a password. Note that if this
+-- column in non-null, this means that user data is server-side
+-- encrypted, not end-to-end-encrypted.
+ALTER TABLE users ADD column encryption_key VARCHAR DEFAULT NULL;
```

Comment:
It makes me a bit nervous to have an encryption key stored in a database in plaintext, particularly in the main users table where it would be hard to segregate later.  I realize this is the case where the user has intentionally opted for less security, so maybe it's not something we need to handle right now, but below are my thoughts on options.

This is aimed at securing against an attacker who manages to access our database (not against client-side attacks).  I think a slightly better version of this would use a single server master key to encrypt this field in the database.  The server master key would then be the kind of thing to store in more secure storage somewhere.  I don't know the latest best-practices around the latter.  Trusted hardware, or a keyfile on disk, or a passphrase required to decrypt at restart, etc.

Reply:
https://github.com/proofcarryingdata/zupass/issues/819